ground/src/main/java/com/google/android/ground/ui/map/MapController.kt

```diff
@@ -43,50 +50,51 @@ constructor(
   private val locationOfInterestRepository: LocationOfInterestRepository,
   private val surveyRepository: SurveyRepository,
   private val mapStateRepository: MapStateRepository,
-  @DefaultDispatcher private val defaultDispatcher: CoroutineDispatcher,
+  @ApplicationScope private val externalScope: CoroutineScope
 ) {
 
-  private val cameraUpdatesSubject: @Hot Subject<CameraPosition> = PublishSubject.create()
+  private val _cameraUpdates = MutableStateFlow<CameraPosition?>(null)
 
   /** Emits a combined stream of camera update requests. */
-  fun getCameraUpdates(): Flowable<CameraPosition> =
-    cameraUpdatesSubject
-      .toFlowable(BackpressureStrategy.LATEST)
-      .mergeWith(getCameraUpdatesFromLocationChanges())
-      .mergeWith(getCameraUpdatedFromSurveyChanges())
+  fun getCameraUpdates(): SharedFlow<CameraPosition> =
+    merge(
+        _cameraUpdates.filterNotNull(),
+        getCameraUpdatesFromLocationChanges(),
+        getCameraUpdatedFromSurveyChanges(),
+      )
+      .shareIn(externalScope, replay = 1, started = SharingStarted.Eagerly)
```

Comment:
Can we simplify this by having the ViewModel collect values from each of these streams and calling the appropriate event method on the Map`panAndZoomCamera`? Possibly out of scope of this PR, but I'd also consider moving this logic into the base ViewModel, since a "Controller" is hard to conceptualize in MVVM architecture model. That might help further reduce the complexity of the map logic.

Reply:
In that case, won't the flow be destroyed and recreated on each screen change?

Reply:
ViewModels survive configuration changes, so as long as we're retrieving the existing one in `onCreate` rather than creating a new one the state should be preserved.

Reply:
Ah, I see. If we're being consistent, view state and behavior should be models in VM. Those states can come from repositories and managers (in our design, the repositories for system states). If we need state to persist across views, we would want to move it upstream to specialized singletons. I'd also consider modeling each usage as a UseCase class if it makes sense. Wdyt?